mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -598,6 +599,17 @@ def after_create(self, block: 'Block', **kwargs) -> None:
                     priority=priority,
                     widget=widget,
                 )
+                if downstream_block_uuids:
```

Comment:
I would add this logic to `pipeline.add_lock` so that we can validate the pipeline circular dependency issues after setting upstream and downstream blocks.

Reply:
Done.